Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -125,44 +132,62 @@ public void clearCache() {
     if (cancellationToken != null && cancellationToken.isCancelled()) {
       return Task.cancelled();
     }
-    final File file = getCacheFile(state);
+    final File cacheFile = getCacheFile(state);
     return Task.call(new Callable<Boolean>() {
       @Override
       public Boolean call() throws Exception {
-        return file.exists();
+        return cacheFile.exists();
       }
     }, Task.BACKGROUND_EXECUTOR).continueWithTask(new Continuation<Boolean, Task<File>>() {
       @Override
       public Task<File> then(Task<Boolean> task) throws Exception {
         boolean result = task.getResult();
         if (result) {
-          return Task.forResult(file);
+          return Task.forResult(cacheFile);
         }
         if (cancellationToken != null && cancellationToken.isCancelled()) {
           return Task.cancelled();
         }
 
-        // network
-        final ParseAWSRequest request = new ParseAWSRequest(ParseRequest.Method.GET, state.url());
+        // Generate the temp file path for caching ParseFile content based on ParseFile's url
+        // The reason we do not write to the cacheFile directly is because there is no way we can
+        // verify if a cacheFile is complete or not. If download is interrupted in the middle, next
+        // time when we download the ParseFile, since cacheFile has already existed, we will return
+        // this incomplete cacheFile
+        final File tempFile = getTempFile(state);
+
+        return Task.call(new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            // If the last download is interrupted, there may be incomplete temp file in disk. So
+            // we need to clear it first.
+            ParseFileUtils.deleteQuietly(tempFile);
+            return null;
```

Comment:
Let's actually remove this part and instead come up with a proper solution to deal with the staging and cache.